### DISCUSSION TITLE
Synology itzg install all ok but fail with mods / plugins

### DISCUSSION THREAD

**Discussion by Knubblo** [itzg/docker-minecraft-server#2035] [Q&A]
Dear community,
Realising that everyone in this community is smarter than me, I cannot code but I can read and understand instructions to some extent:-).
I have successfully installed the server on my Synology, whitelisted access only, RCON disabled and port forward for outside (whitelisted) players. Everything works very fine, no lag, BUT I cannot get my head around how to install mods or plugins, e.g. ATM7 or ATM8. I'll guess I should place the .zip or the files in a or a specific directory but I do not understand where and what additional command line entries I need to input to (e.g. "server.properties")or elsewhere. 

I completely realise that people on Github are way beyond these trivial topics but still, I would be VERY grateful if someone could help out. This is not for me but for my kids. Still I am pretty happy to at least get this far.

Crossing fingers!
Thanks!
/Knubblo

**Comment by itzg** [comment]
Try this mode

https://github.com/itzg/docker-minecraft-server#auto-curseforge-management

where an example of using that for ATM8 is provided here

https://github.com/itzg/docker-minecraft-server/blob/master/examples/atm8/docker-compose.yml

**Reply by Knubblo** [reply]
Thank you for taking the time. Much appreciated. However, one more question if I may?
Where do I put those config instructions? I'll guess not in the "server.properties"-file. Instead the "command line file" / Docker boot parameter settings? Most likely called something else but I'll guess you know what I mean.
Thanks!

**Reply by itzg** [reply]
The example is to be used with Docker compose. Either use an approach like https://linuxhint.com/docker-compose-with-synology/ or translate the variables to the `-e` equivalents for docker run.

**Reply by Knubblo** [reply]
Realising I'm getting very pro help and I appreciate it a lot. Unfortunately I am unable to progress further. I simply do not have enough knowledge about this. 

I have created a yaml file (as per below). When I issue the command "docker-compose up -d" I get the error msg "/docker-compose.yaml', services 'image' must be a mapping not a string.". Having browsed the Net for hours it seems to relate to intendation but I have tried several text editors and also a couple of yaml text verifiers. Perhaps I have not understood what it should read.

Perhaps it was about this "image: itzg/minecraft-server" that needs to be changed to some other text (my config) but I fail to understand what about. Looked through the Docker commands and thought I'd try to list all images but when I enter "docker images" or "docker image ls" I get another error message

"docker: Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/containers/create": dial unix /var/run/docker.sock: connect: permission denied.
See 'docker run --help'."

I cannot understand this since I am doing my SSH via my admin user which is also ticked for R/W of Docker. I have restarted my Synology just in case but the access problem persists.

This is the  yaml-file I entered. I have created a directory "Modpacks" which only contains tha yaml-file which I try to run.

services:
  image: itzg/minecraft-server
  ports:
    - 25565:25565
  environment:
  EULA: "true"
  TYPE: AUTO_CURSEFORGE
  CF_PAGE_URL: https://www.curseforge.com/minecraft/modpacks/all-the-mods-8
  CF_FILENAME_MATCHER: 1.0.6
  MEMORY: 4G
  volumes:
    - /volume1/docker/minecraft/Modpacks
    
    Realising this is probably too much newbie crap for you but anyways:-)
    
    Thanks!

**Reply by itzg** [reply]
The indentation of the `docker-compose.yml` file is important and without the code block delimiters your content seems to have been squashed into one column. Can you edit that with formatting?

The permission issue can be resolved by following the appropriate steps in https://docs.docker.com/engine/install/linux-postinstall/

**Reply by Knubblo** [reply]
Hi,
Thanks. I edited my yaml-file with Notepad (Microsoft Windows 11) and for indents I used spacebar. I counted the spaces in the example yaml-file you posted via link in this thread and I have the same spaces for each row in my file (just went weird when I pasted the textfile in this forum). 
But perhaps "space" is not allowed? Do I need a separate editor and "tab" indents? Having read online I found users running notepad and saving the file as yaml UTF-8. I even tried a few different yaml validators and the all said ok/green light but perhaps they only look at language and commands and not indents?

**Reply by itzg** [reply]
Notepad is fine (but I recommend using a monospace font) and spaces are correct. 

You need to declare a name for the service, such as `mc` [like this](https://github.com/itzg/docker-minecraft-server/blob/master/examples/atm8/docker-compose.yml#L4) and then `image` and others are indented below that. 

I highly recommend looking into using VS Code rather than Notepad since it has extensions that bring awareness of yaml, compose files, etc.

**Reply by Knubblo** [reply]
Ok, I'll try that.

I looked at your suggested yaml-file. Can I use that without changing a thing or do I need to adjust any of the variables? Thinking / guessing that volumes is one of those. Should the image: variable be left as is i.e. "itzg/minecraft-server"

**Reply by itzg** [reply]
Yes, that file could be used as is and that is indeed the image to be using. If you want to use a host directory instead of managed volume, you could replace `mc-data` in line 15 with `./mc-data`

**Reply by Knubblo** [reply]
Well, is was intendation after all. Using VC Code to edit/save the file made the error-msgs go away (notepad goodbye). The access rights was solved with the link you provided. My hat off to you sticking with all my newbie questions. THANKS!

ATM8 was a big beast and it does not look like my NAS can handle it (j218 with 6GB). It runs for a long time, download tonnes of stuff (some 11 pages of log) but the stops with 
![image](https://user-images.githubusercontent.com/129001938/228368722-0605755c-5344-4464-990b-3fff28d2f336.png)

Perhaps there is a little less heavy mods collection that I could try?

**Reply by itzg** [reply]
If it took a while to startup, then it might have been the health check that times it out. There's some info about adjusting that, like increasing retries, [here](https://github.com/itzg/docker-minecraft-server#healthcheck). 

Personally, I prefer no mods at all 😉
